nix/manual-overrides.nix

```diff
@@ -34,30 +37,50 @@ hself: hsuper: {
   binary-parsers = hlib.markUnbroken (hlib.doJailbreak hsuper.binary-parsers);
   bytestring-arbitrary = hlib.markUnbroken (hlib.doJailbreak hsuper.bytestring-arbitrary);
   lens-datetime = hlib.markUnbroken (hlib.doJailbreak hsuper.lens-datetime);
-  network-arbitrary = hlib.markUnbroken (hlib.doJailbreak hsuper.network-arbitrary);
-  proto-lens-protoc = hlib.doJailbreak hsuper.proto-lens-protoc;
-  proto-lens-setup = hlib.doJailbreak hsuper.proto-lens-setup;
-  th-desugar = hlib.doJailbreak hsuper.th-desugar;
+  libsodium = hlib.markUnbroken (hlib.addPkgconfigDepend hsuper.libsodium (
+    libsodium.overrideAttrs (old:
+      rec {
+        # we don't care for the patches for mingw and for 1.0.19
+        patches = [ ];
+        version = "1.0.18";
+        src = fetchurl {
+          url = "https://download.libsodium.org/libsodium/releases/${old.pname}-${version}.tar.gz";
+          hash = "sha256-b1BEkLNCpPikxKAvybhmy++GItXfTlRStGvhIeRmNsE=";
+        };
+      }
+    )));
```

Comment:
Why not put this in haskell-pins.nix?

Reply:
Also, it would be helpful to document what problems 1.0.19 causes.

Reply:
> Why not put this in haskell-pins.nix?

because the library itself doesn't change, it's just the actual `libsodium` that gets pinned - `libsodium` the c library has moved on upstream but many (all) haskell libraries are only compatible with 1.0.18.